integration_tests/src/main/python/limit_test.py

```diff
@@ -56,6 +56,7 @@ def test_non_zero_offset(offset):
 
 @pytest.mark.parametrize('limit, offset', [(0, 0), (0, 10), (1024, 500), (2048, 456), (3000, 111), (500, 500), (100, 600)])
 @pytest.mark.skipif(is_before_spark_340(), reason='offset is introduced from Spark 3.4.0')
+@allow_non_gpu('ShuffleExchangeExec') # when limit = 0, ShuffleExchangeExec is not replaced.
```

Comment:
This is because the Spark will optimize the logical plan by replacing the child plan with `empty table`.
On Spark 3.4: 
```scala
scala> spark.sql("create table tt4 as select 1 as a")
22/11/29 10:15:56 WARN ResolveSessionCatalog: A Hive serde table will be created as there is no table provider specified. You can set spark.sql.legacy.createHiveTableByDefault to false so that native data source table will be created instead.
res0: org.apache.spark.sql.DataFrame = []

scala> spark.sql("select * from tt4 limit 0").repartition(1).explain(true)
== Parsed Logical Plan ==
Repartition 1, true
+- GlobalLimit 0
   +- LocalLimit 0
      +- Project [a#0]
         +- SubqueryAlias spark_catalog.default.tt4
            +- HiveTableRelation [`spark_catalog`.`default`.`tt4`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#0], Partition Cols: []]

== Analyzed Logical Plan ==
a: int
Repartition 1, true
+- GlobalLimit 0
   +- LocalLimit 0
      +- Project [a#0]
         +- SubqueryAlias spark_catalog.default.tt4
            +- HiveTableRelation [`spark_catalog`.`default`.`tt4`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#0], Partition Cols: []]

== Optimized Logical Plan ==
Repartition 1, true
+- LocalRelation <empty>, [a#0]

== Physical Plan ==
Exchange SinglePartition, REPARTITION_BY_NUM, [plan_id=6]
+- LocalTableScan <empty>, [a#0]
```

When running on spark-rapids,  the `ShuffleExchangeExec` won't be replaced because the child is empty:
```scala
scala> spark.sql("select * from tt4 limit 0").repartition(1).explain
22/11/29 10:26:20 WARN GpuOverrides: 
!Exec <ShuffleExchangeExec> cannot run on GPU because Columnar exchange without columnar children is inefficient
  @Partitioning <SinglePartition$> could run on GPU
  ! <LocalTableScanExec> cannot run on GPU because GPU does not currently support the operator class org.apache.spark.sql.execution.LocalTableScanExec
    @Expression <AttributeReference> a#14 could run on GPU

== Physical Plan ==
Exchange SinglePartition, REPARTITION_BY_NUM, [plan_id=153]
+- LocalTableScan <empty>, [a#14]
```